cmd/sriov-network-config-daemon/start.go

```diff
@@ -109,6 +109,10 @@ func runStartCmd(cmd *cobra.Command, args []string) error {
 
 	vars.ParallelNicConfig = startOpts.parallelNicConfig
 
+	if strings.ToLower(os.Getenv("MLX_PLUGIN_FW_RESET")) == "true" {
```

Comment:
Some questions:
1) Why are we using ENV variables? When there are other means of getting configuration data e.g. config map
2) Why is this even configurable? Shouldn't NVIDIA know how to detect when their hardware needs to be reset a certain way? It seems like having the user defined this ENV is not a good UX.

Reply:
Regarding your questions:

1. we discussed this in the office hours today and we have come to the conclusion that adding env vars is a very generic and non-disruptive way. We could also use a ConfigMap for this, but I'd suggest to implement this change in a follow-up PR and move all configuration to the ConfigMap.
2. This change is only necessary for certain cards, not for all cards (AFAIK). If we are rebooting the host anyway, isn't it okay to reset the FW as well? WDYT @adrianchiris @ykulazhenkov

Reply:
1. @wizhaoredhat  you attended the community meeting IIRC. This was the chosen approach.

2. its not related to Nvidia Hardware its related to the server (Intel's 4th generation Xeon "Eagle Stream") and how it performs PCIe reset on boot. im not aware of a way to detect it. @SchSeba has more details.

Reply:
> Yes I did attend the meeting. But this comment was made taking some time and thinking about it more.

OK, my take,  lets not add another object (CM) into the mix which needs to be created/cleaned-up, at this point its not worth it.

> Can't we run lspci and grep for the CPU? And make decisions based on the generation of Intel processors? Maybe there is a instruction that is no longer exposed for their processor and there is a sysfs file describing that. Just thinking out loud.

Im open to concrete suggestions. im not aware of such way. i think eventually we can enable it by default. but definitely not for the first iteration.

please touch base with Sebastian he can provide you with the relevant BZ if you need more context.

Reply:
Sorry I apologize that I don't know all the details of the issue. I will reach out to Sebastian.

Why can't we enable it by default now? Then we can skip the feature flag entirely. Is there a side effect of the reset that the behaviour is not well defined?

Reply:
Implemented in https://github.com/k8snetworkplumbingwg/sriov-network-operator/pull/733/commits/2c4dfe01ff0ae55a2f9e0edcba3699b69d08dfd0. Thanks for the hint @SchSeba